### ISSUE TITLE
S1125 suggested action does not match what the rule is checking to prevent

### ISSUE THREAD

**Issue by StingyJack** [SonarSource/sonar-dotnet#3407]
Given that the function `IsConditionPresent` returns a boolean value and the code snippet
``` csharp
if (IsConditionPresent() == false) 
{
    DoSomething();
}
```

Rule S1225 (Boolean literals should not be redundant) suggests changing out the boolean literal for a negation operator so it looks like instead
``` csharp
if (!IsConditionPresent()) 
{
    DoSomething();
}
```

I would expect that rule to show if the code had been 
``` csharp
if (IsConditionPresent() == true) 
{
    DoSomething();
}
```
... because per the rule text, the boolean is redundant and logically no different from
``` csharp
if (IsConditionPresent()) 
{
    DoSomething();
}
```
The suggestion its making in the `== false` case is not inline with redundant boolean literals. Its suggesting an alternate style but not actually removing any redundancy, its just moving it to the left a bit. This suggested representation - while popular - has led to more than one bug Ive had to deal with because its harder to see `(!IsSomethingTrue())` than it is to see `(IsSomethingTrue() == false)`, especially with how eyesight usually deteriorates with age. 

I'd prefer that this rule be left as is for the redundant case, but move the negation movement evaluation to its own rule. This would let me have the benefit of the redundancy identification, but could turn off the negation movement styling.

**Comment by pavel-mikula-sonarsource** [comment]
Hi @StingyJack ,

Thank you for raising the point. The code style here is generally questionable and the result depends on personal, project and context preference. We've recently discussed this `== false` topic in S1125 and the conclusion was to keep the rule as is.

You can disable the rule or mark the cases as "Won't fix" in your SonarQube/SonarCloud.

**Comment by StingyJack** [comment]
> The code style here is generally questionable

@pavel-mikula-sonarsource - I can tell you that It becomes less questionable as one's eyesight diminishes. I used to use negation until I had to work with a team of developers where the average age was over 40, and there were more than one defect related to an `!` being missed in an `if (!IsReady)`.  Writing `if (IsReady == false)` instead was easier to read and doesnt carry the risk of missing a negation operator that sandwiched between `(` and often followed by `I`.
